src/target/riscv/riscv.c

```diff
@@ -1953,7 +1953,8 @@ static int riscv_run_algorithm(struct target *target, int num_mem_params,
 
 	/* Write memory parameters to the target memory */
 	for (int i = 0; i < num_mem_params; i++) {
-		if (mem_params[i].direction != PARAM_IN) {
+		if (mem_params[i].direction == PARAM_OUT ||
+				mem_params[i].direction == PARAM_IN_OUT) {
```

Comment:
I don't understand how this change has an effect. param_direction only defines IN, OUT, and IN_OUT. So checking for "not IN" is the same as "OUT or IN_OUT", right? What am I missing?

Reply:
> I don't understand how this change has an effect

This is right, this change has kind of no behavioural effect.

> What am I missing?

I changed it to be more in line with the rest of the code base. Ex : 
https://github.com/riscv/riscv-openocd/blob/4fb152f733651a5461080f2ebe65eb234bf401b1/src/target/riscv/riscv.c#L2073

Reply:
Oh, I clicked the wrong link in github and all it showed me was this one cleanup, not the functional part of your code.

Reply:
Ahhhh XD, got it ^^